views/components/ConfigurationsList/index.jsx

```
@@ -0,0 +1,840 @@
+        [props.selectProps.classes.chipFocused]: props.isFocused,
+      })}
+      onDelete={props.removeProps.onClick}
+      deleteIcon={<CancelIcon {...props.removeProps} />}
+    />
+  )
+}
+
+function Menu(props) {
+  return (
+    <Paper
+      square
+      className={props.selectProps.classes.paper}
+      {...props.innerProps}
+    >
+      <MenuList>{props.children}</MenuList>
+    </Paper>
+  )
+}
+
+const components = {
+  Control,
+  Menu,
+  MultiValue,
+  NoOptionsMessage,
+  Option,
+  Placeholder,
+  SingleValue,
+  ValueContainer,
+}
+const ConfigurationsList = ({ user, classes, theme }) => {
+  const [state, setState] = useState({
+    user: {},
+    preferences: {},
+    configurations: [],
+    gridCols: 1,
+    gridWidth: 350,
+    searchUsers: false,
+    friends: [],
+    shared: [],
+    snackTime: false,
+    uploadSnack: false,
+    selectedConfig: {},
+    configOwner: '',
+    totalStudies: 0,
+    totalSubjects: 0,
+    totalDays: 0,
+  })
+
+  useEffect(() => {
+    loadUserNames()
+    handleResize()
+    fetchConfigurations(user.uid).then((configurations) => {
+      setState((prevState) => {
+        return {
+          ...prevState,
+          configurations,
+        }
+      })
+    })
+    fetchPreferences(user.uid)
+    window.addEventListener('resize', handleResize)
+
+    return () => window.removeEventListener('resize', handleResize)
+  }, [])
+
+  useEffect(() => {
+    if (user.message.length > 0) {
+      setState((prevState) => {
+        return {
+          ...prevState,
+          uploadSnack: true,
+        }
+      })
+    }
+  }, [user])
+
+  const loadUserNames = async () => {
+    try {
+      const usernames = await fetchUsernames()
+      setState((prevstate) => {
+        return {
+          ...prevstate,
+          friends: usernames.map((username) => ({
+            value: username,
+            label: username,
+          })),
+        }
+      })
+    } catch (error) {}
```

Do we want to swallow errors or alert ourselves to them?

---

[74ec429](https://github.com/AMP-SCZ/dpdash/pull/304/commits/74ec4294f8513a150c7fb60f543504bd28591100)